src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -289,11 +293,34 @@ private CollectionExercise collectionExerciseTransitionState(CollectionExercise
 
     try {
       if (published == exercise.getSampleSize().longValue()) {
-        // All sample units published, set exercise state to PUBLISHED
-        exercise.setState(
-            collectionExerciseTransitionState.transition(
-                exercise.getState(), CollectionExerciseDTO.CollectionExerciseEvent.PUBLISH));
-        exercise.setActualPublishDateTime(new Timestamp(new Date().getTime()));
+
+        if ((eventRepository
```

Comment:
1. Extracting this out into a variable would improve the readability i.e. isGoLiveTime
2. What if a go live event hasn't been loaded?

Reply:
@benjefferies A collection exercise can't get into the validated state if it does not have all mandatory events loaded. These are required for a collection exercise to go into ready for review to my understanding.